charts/datadog/templates/_containers-common-env.yaml

```diff
@@ -150,4 +151,32 @@ Return a list of env-vars if the cluster-agent is enabled
         name: {{ .Values.existingClusterAgent.tokenSecretName | quote }}
         key: token
 {{- end }}
+
+# Dogstatsd configuration
```

Comment:
some of these parameters are only use by the dogstatsd server, so why do we want to add them to all containers?

Reply:
As part of our OKRs we want to align the configuration between all containers. The goal is for each process/container to see the same configuration. That will solve the discrepancy in the metadata we see in the backend (in remote-config and fleet automation).
Main use case, for now, is for the core-agent to send metadata about other products accurately. For example, with our helm chart, the core-agent doesn't have access to the actual system-probe config. We therefore send the wrong metadata to the backend.

Reply:
the main issue that I have with this approach is that we blowup the other processes with useless envvars. 
For the agent deployment in containerise environment we try to follow the cloud native app best practice: https://12factor.net/

Reply:
This replicates what we do with other deployment methods (Ansible, Chef, DEB, ...) where all processes share the same configuration file and have the same state when it comes to configuration/settings.

Having the same behavior for containers allow us to use the same logic for both backend and agent side with containers deployments. I don't think it's a workaround since it aligns the agents behavior to be the same than in non-k8s world.
Also, I would not call a few envars  a blowup, were only talking about the configuration.